query_generators/openai.py

```diff
@@ -143,16 +168,45 @@ def generate_query(
                     columns_to_keep,
                     shuffle,
                 )
+                table_metadata_ddl = convert_ddl(
+                    postgres_ddl=table_metadata_ddl,
+                    to_dialect=self.db_type,
+                    db_name=self.db_name,
+                )
                 table_metadata_string = table_metadata_ddl + join_str
             elif columns_to_keep == 0:
                 md = dbs[self.db_name]["table_metadata"]
-                table_metadata_string = to_prompt_schema(md, shuffle)
+                table_metadata_ddl = to_prompt_schema(md, shuffle)
+                table_metadata_ddl = convert_ddl(
+                    postgres_ddl=table_metadata_ddl,
+                    to_dialect=self.db_type,
+                    db_name=self.db_name,
+                )
```

Comment:
same issue raised above of adding translation after pruning/non-pruning

Reply:
Will fix!